lib/src/client/channel.dart

```diff
@@ -47,6 +48,8 @@ abstract class ClientChannelBase implements ClientChannel {
 
   bool _isShutdown = false;
 
+  static bool enableTimelineLogging = false;
```

Comment:
I would remove this field from here and instead have it in the `profiler.dart` as a public member

```dart
bool isTimelineLoggingEnabled = false;
```

also note the naming (`enable` is a verb so it is unsuitable for variable names).